src/libostree/ostree-repo-commit.c

```diff
@@ -734,6 +734,17 @@ write_content_object (OstreeRepo         *self,
     }
   else
     {
+      /* Update size metadata if configured */
```

Comment:
Surely the important thing, rather than that they are mutually exclusive, would be whether they are exhaustive? (Mutually exclusive: !(a && b) always; exhaustive: (a || b) always)

There are three cases:

* G_FILE_TYPE_REGULAR (phys_object_is_symlink is false)
* G_FILE_TYPE_SYMBOLIC_LINK && mode in (BARE, BARE_USER_ONLY) (phys_object_is_symlink is true)
* G_FILE_TYPE_SYMBOLIC_LINK && mode not in (BARE, BARE_USER_ONLY) (phys_object_is_symlink is false)

By getting here we know we are in one of the two !phys_object_is_symlink cases, so not the second one; but we don't know whether it is the first or the third. So the check seems non-redundant to me.